src/Microsoft.AspNetCore.Server.Kestrel/Http/SocketInput.cs

```diff
@@ -180,25 +180,29 @@ public MemoryPoolIterator ConsumingStart()
 
             lock (_sync)
             {
-                if (!consumed.IsDefault)
+                // object was disposed
```

Comment:
nit: make this comment more informative e.g. "Make sure object was not disposed", "Check if object was disposed" or "Object may have been disposed"